### PR TITLE
Fixes and updates for thermostat and display:

### PR DESCRIPTION
**Pull request by ElderJoy**
- Switch to original esp8266-oled-ssd1306 library
- Fixes and updates for thermostat and display
- Add display switching off after interval
- If THERMOSTAT_DISPLAY_SUPPORT enabled, then one click enable display, long click switch relay. This functionality also depend on fix for long click. See pull request https://github.com/xoseperez/espurna/pull/2172

### CODE REVIEW

File: code/espurna/thermostat.h [hunk 1]
```diff
@@ -13,6 +13,12 @@ Copyright (C) 2017 by Dmitry Blinov <dblinov76 at gmail dot com>
 
 #if THERMOSTAT_DISPLAY_SUPPORT
 #include <SSD1306.h> // alias for `#include "SSD1306Wire.h"`
+
+// It looks more natural that one click will enable display
+// and long click will switch relay
+#define BUTTON1_CLICK           BUTTON_MODE_DISPLAY_ON
+#define BUTTON1_LNGCLICK        BUTTON_MODE_TOGGLE
```
Comment (mcspr) [inline code/espurna/thermostat.h]: Not showing in CI, as display is off by default:
```cpp
In file included from espurna/pr2173/code/espurna/espurna.ino:62:0:
espurna/thermostat.h:19:0: warning: "BUTTON1_CLICK" redefined [enabled by default]
 #define BUTTON1_CLICK           BUTTON_MODE_DISPLAY_ON
 ^
In file included from espurna/config/all.h:36:0,
                 from espurna/pr2173/code/espurna/espurna.ino:22:
espurna/config/defaults.h:87:0: note: this is the location of the previous definition
 #define BUTTON1_CLICK       BUTTON_MODE_TOGGLE
 ^
In file included from espurna/pr2173/code/espurna/espurna.ino:62:0:
espurna/thermostat.h:20:0: warning: "BUTTON1_LNGCLICK" redefined [enabled by default]
 #define BUTTON1_LNGCLICK        BUTTON_MODE_TOGGLE
 ^
In file included from espurna/config/all.h:36:0,
                 from espurna/pr2173/code/espurna/espurna.ino:22:
espurna/config/defaults.h:162:0: note: this is the location of the previous definition
 #define BUTTON1_LNGCLICK    BUTTON_MODE_RESET
 ^
/
```

This should probably be in dependencies.h, so we already have these values after all.h is loaded

Comment (ElderJoy) [reply code/espurna/thermostat.h]: Moved button definitions into the right place. Checked on default configuration and with display support enabled. Please verify that it don't cause conflicts any more. Please note, that thermostat support should be enabled also in this case as display require it.

File: code/platformio.ini [hunk 2]
```diff
@@ -132,7 +132,7 @@ lib_deps =
     https://github.com/sparkfun/SparkFun_VEML6075_Arduino_Library#V_1.0.3
     https://github.com/pololu/vl53l1x-arduino#1.0.1
     https://github.com/mcleng/MAX6675-Library#2.0.1
-    https://github.com/ElderJoy/esp8266-oled-ssd1306#4.0.1
+    https://github.com/ThingPulse/esp8266-oled-ssd1306
```
Comment (mcspr) [inline code/platformio.ini]: pin https://github.com/ThingPulse/esp8266-oled-ssd1306/commit/3398c975a6fb058b3a75c704d9f6ebf1312a97f0 or 4.1.0?

Comment (ElderJoy) [reply code/platformio.ini]: 4.1.0 - tag in my fork of this lib. It is not valid in the original repo.
I tested original library on it's HEAD commit and it works fine for me. So, let's pin it to latest commit #3398c97.

File: code/espurna/thermostat.h (hunk 1 above)
Comment (mcspr) [reply code/espurna/thermostat.h]: Not seeing any more issues with code/test/build/sensor.h config + display flag, lgtm

**Pull request merged**